### PR TITLE
Documenting Array.cr

### PR DESCRIPTION
**Pull request by arktisklada**

### CODE REVIEW

File: src/array.cr [hunk 1]
```diff
@@ -1115,12 +1112,18 @@ class Array(T)
     self
   end
 
-  # Equivalent to `Array#select` but makes modification on the current object rather that returning a new one. Returns nil if no changes were made
+  # Modifes `self`, keeping only the elements in the collection for which the passed block returns `true`.
+  # Returns nil if no changes were made.
+  #
+  # Equivalent to `Array#select`
```
Comment (asterite) [inline src/array.cr]: Maybe "See also" instead of "Equivalent", because they are not equivalent (same goes with reject)

File: src/array.cr [hunk 2]
```diff
@@ -1483,10 +1486,12 @@ class Array(T)
     self
   end
 
+  # Returns an array with all the elements in the collection reversed.
   def reverse
     Array(T).new(size) { |i| @buffer[size - i - 1] }
   end
 
+  # Modifies `self` with all the elements in the collection reversed.
```
Comment (asterite) [inline src/array.cr]: Maybe "Reverses in-place the elements of `self`"?

File: src/array.cr [hunk 3]
```diff
@@ -1630,15 +1648,31 @@ class Array(T)
...
   end
 
+  # Modifies `self` by randomizing the order of elements in the collection
+  # using the given *random* number generator.
   def shuffle!(random = Random::DEFAULT)
     @buffer.shuffle!(size, random)
     self
   end
 
+  # Returns an array with all the elements in the collection sorted using *quicksort*.
```
Comment (asterite) [inline src/array.cr]: I don't know if I would mention the sorting algorithm. I wouldn't mention `Array.quicksort!`, it's just an implementation detail and it's not public API (we should mark it as protected).

**Comment by asterite** [comment]
Thank you! Overall really good changes, talking about "`self` and `other`" vs. "the two arrays" or "ary" (which doesn't exist in the code) is much better in my opinion.

File: src/array.cr [hunk 4]
```diff
@@ -1115,12 +1112,18 @@ class Array(T)
     self
   end
 
-  # Equivalent to `Array#select` but makes modification on the current object rather that returning a new one. Returns nil if no changes were made
+  # Modifes `self`, keeping only the elements in the collection for which the passed block returns `true`.
```
Comment (tristil) [inline src/array.cr]: s/Modifes/Modifies

File: src/array.cr (hunk 1 above)
Comment (arktisklada) [inline src/array.cr]: :+1:

File: src/array.cr (hunk 2 above)
Comment (arktisklada) [inline src/array.cr]: :+1:

File: src/array.cr (hunk 3 above)
Comment (arktisklada) [inline src/array.cr]: :+1:

**Comment by arktisklada** [comment]
@asterite thanks for the feedback!  I made some adjustments, let me know what you think.  Also I noticed a discrepancy in calling out argument names:  Sometimes they're written with * and others with ` (i.e. *random* vs `random`).  I feel the backticks ` are more commonly understand as code.  Thoughts?

File: src/array.cr (hunk 4 above)
Comment (arktisklada) [inline src/array.cr]: :+1: great catch

**Comment by arktisklada** [comment]
Also, are we abiding by an 80-char column limit or something like that?

**Comment by asterite** [comment]
@arktisklada We have the convention to use `*arg*` for arguments. If you use backticks and the name is a method, the doc generator will create a link to that method. So please change all arguments to use stars instead of backticks (it's OK to use backticks with `self`).

The 80-char column limit is something from the past, nowadays monitors are bigger, editors have word-wrap, etc.

**Comment by arktisklada** [comment]
@asterite (or whomever appropriate) is this PR too big?  If I were to add more code-level documentation, would it be best broken into smaller pieces?

**Comment by asterite** [comment]
@arktisklada Oh, it's just that I didn't realize you made the changes above. I'll take a look into this in these days.

**Comment by asterite** [comment]
@arktisklada Looks really good :-)

Just two things:
* There are a couple of `nil` between starts (`*nil*`) that should be between backticks (``` `nil` ```).
* Instead of "This expression returns" use "This method returns".

**Comment by arktisklada** [comment]
@asterite awesome, thanks!  Made those updates: *nil* => `nil` and expression => method

**Comment by arktisklada** [comment]
......and rebased

**Comment by asterite** [comment]
Thank you!! :smile_cat:

**Pull request merged**